contrib/babelfishpg_tsql/src/pltsql_bulkcopy.c

```diff
@@ -943,12 +943,12 @@ BeginBulkCopy(Relation rel,
  * EndBulkCopy - Clean up storage and release resources for BULK COPY.
  */
 void
-EndBulkCopy(BulkCopyState cstate)
+EndBulkCopy(BulkCopyState cstate, bool aborted)
 {
 	if (cstate)
 	{
 		/* Flush any remaining bufferes out to the table. */
-		if (!CopyMultiInsertInfoIsEmpty(&cstate->multiInsertInfo))
+		if (!CopyMultiInsertInfoIsEmpty(&cstate->multiInsertInfo) && !aborted)
```

Comment:
Quick Question:
`CopyMultiInsertInfoFlush()` function does other cleanup like resetting `miinfo->bufferedTuples = 0`, etc.
We are skipping that now in case of an error, is that safe to do?

Reply:
Yeah, its safe to do so.